sarracen/tests/test_render.py

```diff
@@ -64,6 +65,34 @@ def test_1dcross_plot():
     assert ax.get_ylim() == (0, CubicSplineKernel().w(np.sqrt(41) / 1000, 2))
 
 
+def test_3d_plot():
+    df = pd.DataFrame({'rx': [0, 5],
+                       'P': [1, 1],
+                       'h': [1, 1],
+                       'rz': [3, 1],
+                       'rho': [1, 1],
+                       'y': [0, 4],
+                       'm': [1, 1]})
+    sdf = SarracenDataFrame(df)
+
+    fig, ax = sdf.render_3d('P', pixcountx=300, int_samples=10000)
+
+    assert isinstance(fig, Figure)
+    assert isinstance(ax, Axes)
+
+    assert ax.get_xlabel() == 'rx'
+    assert ax.get_ylabel() == 'y'
+    # the colorbar is contained in a second axes object inside the figure
+    assert fig.axes[1].get_ylabel() == 'column P'
+
+    assert ax.get_xlim() == (0, 5)
+    assert ax.get_ylim() == (0, 4)
+    # particle width is 1/60
+    # therefore closest pixel has q=sqrt(2*(1/120)^2)
+    # F(q) ~= 0.477372919027 (calculated externally)
+    assert fig.axes[1].get_ylim() == (0, approx(0.477372919027))
```

Comment:
Try calculating this value in the code as much as you can. It's valuable because it shows your workings. I know you have comments explaining it (keep them!), but code is repeatable and it will make it easier to tweak the test if we ever need to.